### PR TITLE
Tile expiration in the server

### PR DESCRIPTION
**Pull request by lilleyse**
In testing https://github.com/AnalyticalGraphicsInc/cesium/pull/4136 I edited `server.js` to serve different tiles when `points.pnts` is requested, but I'm getting some strange results. It looks like the data is getting corrupted after a certain point, somewhere in the middle of the positions buffer (and continuing onto the colors buffer). Otherwise the tile is the correct byte size. I have also checked files individually and they are correct, so something must be wrong or missing from my server code.

Corrupted:
corrupted

Correct:
not-corrupted

### CODE REVIEW

**Comment by lilleyse** [comment]
@mramato do you have any hunches about this? The server.js code is pretty short.

**Comment by mramato** [comment]
Are you sure it's getting into your new handler?  The `gzipAndNext` handler is registered first and it is set up to handle `pnts` as well, this would preclude your handler from getting called.

**Comment by lilleyse** [comment]
I think that part was ok because the next() would call the next middleware, which would be the expire tileset checker. At least, both areas print to the console.

This did lead me to the problem though, which was I wasn't being careful about the ordering of the middlewares. `app.use(express.static(__dirname));` was called last, and probably clobbered the `res.sendFile` that I had. I had to shuffle around some things and just not call `next()` in the expire tileset checker.

**Comment by lilleyse** [comment]
I'm probably wrong about some of that terminology, and I'm actually not sure why app.get is allowed to be handled twice. It seems like all the `.get` should actually be `.use`.

**Comment by pjcozzi** [comment]
Given that we want to merge https://github.com/AnalyticalGraphicsInc/cesium/pull/4136, what's left here?

**Comment by lilleyse** [comment]
This is ready. All the issues above are sorted out.

**Comment by pjcozzi** [comment]
@mramato could you take a quick look at this and merge?

File: server.js [hunk 1]
```diff
@@ -74,12 +81,19 @@
...
+    var expirationPntsPath = '/tilesets/TilesetWithExpiration/points.pnts';
+    var expirationCacheDirectory = '/tilesets/TilesetWithExpiration/cache/';
+    var expirationCacheLength = 5;
+    var expireCount = 0;
+
+    app.use(expirationPntsPath, function(req, res) {
+        var pntsPath = expirationCacheDirectory +  'points_' + expireCount + '.pnts';
+        expireCount = (expireCount + 1) % expirationCacheLength;
+        res.sendFile(pntsPath, {root: __dirname});
+        // Don't call next() because we don't need to run the express.static middleware
```
Comment (mramato) [inline server.js]: I think the more correct thing to do is use `app.get` here and always call next.  Similarly, `express.static` should be an `app.get(/.*/, express.static(__dirname));)`.

`app.get` is applied "waterfall" style, so the earliest `app.get` that matches a url wins and then processing ends.
`app.use` is run on everything, which clearly is not what we want for these requests.

If you try these changes and run into problems, we can just punt for now and I'll write an issue to revisit.

Comment (lilleyse) [inline server.js]: > app.get is applied "waterfall" style, so the earliest app.get that matches a url wins and then processing ends.

I'm not sure if this is actually what happens. Both these get called:

```
    app.get(expirationPntsPath, function(req, res, next) {
        console.log('first');
        next();
    });

    app.get(/.*/, function(req, res, next) {
        console.log('second');
        next();
    });
```

Comment (mramato) [inline server.js]: Totally not what I expect it, but I believe it.  Not sure if express and restify have different semantices and I'm more used to rest.  We can just keep this as-is for now then.

**Comment by mramato** [comment]
> @mramato could you take a quick look at this and merge?

I actually don't have commit access, so someone else needs to hit merge.

**Pull request merged**